# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1412054770 18000
#      Tue Sep 30 00:26:10 2014 -0500
# Branch stable
# Node ID d60b44acfff765ebd061b0ac1ab97ec482548c5b
# Parent  f96495e88a70e54694830b5c5f3b01f79c5beecc
Change a bitwise OR to a logical OR in Range.h.

* Range.h (Range::Range): Replace | with || in logical test that snuck in
during cset c457a84bc7d3 when improving range/scalar arithmetic.

diff --git a/liboctave/array/Range.h b/liboctave/array/Range.h
--- a/liboctave/array/Range.h
+++ b/liboctave/array/Range.h
@@ -49,17 +49,17 @@ public:
     : rng_base (b), rng_limit (l), rng_inc (i),
       rng_nelem (nelem_internal ()), cache () { }
 
   // For operators' usage (to preserve element count).
   Range (double b, double i, octave_idx_type n)
     : rng_base (b), rng_limit (b + (n-1) * i), rng_inc (i),
       rng_nelem (n), cache ()
   {
-    if (! xfinite (b) || ! xfinite (i) | ! xfinite (rng_limit))
+    if (! xfinite (b) || ! xfinite (i) || ! xfinite (rng_limit))
       rng_nelem = -2;
   }
 
   double base (void) const { return rng_base; }
   double limit (void) const { return rng_limit; }
   double inc (void) const { return rng_inc; }
   octave_idx_type nelem (void) const { return rng_nelem; }
 
